### ISSUE TITLE
Integrate a validation set

### ISSUE THREAD

**Issue by Vectorrent** [0-5788719150923125/ode#2]
The challenge is with how to split the data, when the vast majority of it is being sampled stochastically, today.

**Comment by Vectorrent** [comment]
I added some very basic (and kind of ugly) validation code. The only data that supports it currently is the Phi dataset. 

Because Phi has just 2 batches, I assigned one to training data, and the other to validation data. 

I also implemented a simple, deterministic random number generator, so that I could initialize weights and iterate over data in a reproducible fashion.

There are likely to be things I missed, so we'll leave this issue open for a while.
